src/cultionet/models/nunet.py

```diff
@@ -921,14 +1184,11 @@ def forward(
             x4_0=x4_0
         )
 
-        dist = self.final_dist(out_0_4['dist'])
-        edge = self.final_edge(out_0_4['edge'])
-        mask = self.final_mask(out_0_4['mask'])
-
-        out = {
-            'dist': dist,
-            'edge': edge,
-            'mask': mask
-        }
+        out = self.post_unet(
+            out_0_4=out_0_4,
+            out_3_1=out_3_1,
+            out_2_2=out_2_2,
+            out_1_3=out_1_3
+        )
 
         return out
```

Comment:
This file does seem verbose, with a lot of repeated blocks. Attempting to reduce repetition could be the subject of a future PR.